compiler/rustc_middle/src/mir/consts.rs

```diff
@@ -463,6 +463,9 @@ impl<'tcx> Const<'tcx> {
                 let const_val = tcx.valtree_to_const_val((ty, valtree));
                 Self::Val(const_val, ty)
             }
+            ty::ConstKind::Unevaluated(ty::UnevaluatedConst { def, args }) => {
```

Comment:
This is not correct, and there's no justification here? What is this for?

Reply:
Specifically, there's a distinction between `ty::ConstKind::Unevaluated(ty::UnevaluatedConst(..))` and `mir::ConstKind::Unevaluated(..)` and I think this may be getting that wrong. But generally randomly changing a function to do something different without asking why it's that way that seems like it may introduce subtle bugs 🤔